src/providers/wfs/qgswfsprovidermetadata.cpp

```diff
@@ -0,0 +1,365 @@
+  details.setType( QgsMapLayerType::VectorLayer );
+  details.setProviderKey( QgsWFSProvider::WFS_PROVIDER_KEY );
+  details.setUri( uri );
+  details.setName( wfsUri.typeName() );
+  details.setWkbType( provider.wkbType() );
+  res << details;
+
+  if ( wfsUri.hasGeometryTypeFilter() ||
+       !caps.supportsGeometryTypeFilters() )
+  {
+    if ( provider.wkbType() == QgsWkbTypes::Type::Unknown )
+      provider.issueInitialGetFeature();
+    details.setWkbType( provider.wkbType() );
+    return res;
+  }
+
+  if ( provider.wkbType() == QgsWkbTypes::Type::Unknown &&
+       provider.sharedData()->layerProperties().size() == 1 )
+  {
+    std::vector<std::unique_ptr<QgsWFSGetFeature>> requests;
+    std::set<QgsWFSGetFeature *> finishedRequests;
+    constexpr int INDEX_ALL = 0;
+    constexpr int INDEX_NULL = 1;
+    constexpr int INDEX_POINT = 2;
+    constexpr int INDEX_CURVE = 3;
+    constexpr int INDEX_SURFACE = 4;
+    // Order of strings in the list must be consistent with the INDEX_* enumeration above
+    const QStringList filterNames = { QString(), // all features
+                                      QString( "IsNull" ),
+                                      QStringLiteral( "IsPoint" ),
+                                      QStringLiteral( "IsCurve" ),
+                                      QStringLiteral( "IsSurface" )
+                                    };
+
+    const auto downloaderLambda = [&]()
+    {
+      QEventLoop loop;
+      QTimer timerForHits;
+      for ( QString function : filterNames )
+      {
+        QString filter;
+        if ( function == QLatin1String( "IsNull" ) )
+        {
+          filter = QgsWFSProvider::buildIsNullGeometryFilter( caps, provider.geometryAttribute() );
+        }
+        else if ( !function.isEmpty() )
+        {
+          filter = QgsWFSProvider::buildFilterByGeometryType( caps,
+                   provider.geometryAttribute(),
+                   function );
+        }
+
+        if ( !provider.sharedData()->WFSFilter().isEmpty() )
+        {
+          filter = provider.sharedData()->combineWFSFilters( {filter, provider.sharedData()->WFSFilter()} );
+        }
+
+        requests.emplace_back( std::make_unique<QgsWFSGetFeature>( wfsUri ) );
+
+        requests.back()->request( /* synchronous = */ false,
+            caps.version,
+            wfsUri.typeName(),
+            filter,
+            /* hitsOnly = */ true,
+            caps );
+
+        QgsWFSGetFeature *thisRequest = requests.back().get();
+        const auto downloadFinishedLambda = [ &, thisRequest]()
+        {
+          finishedRequests.insert( thisRequest );
+          if ( finishedRequests.size() == requests.size() )
+            loop.quit();
+        };
+        connect( thisRequest, &QgsWfsRequest::downloadFinished, thisRequest, downloadFinishedLambda );
+      }
+      QgsSettings s;
+      const double timeout = s.value( QStringLiteral( "qgis/wfsGetFeatureGeometryTypesTimeout" ), 2.0 ).toDouble();
```

Comment:
I can't spot `qgis/wfsGetFeatureGeometryTypesTimeout` in advanced settings to customize it. It's also not listed in QGIS3.ini.

Reply:
addressed in 761f0292597d70118145a7aed0857d456a811099

Reply:
:+1: